tarantool_test.go

```diff
@@ -3148,6 +3156,91 @@ func TestConnectionFeatureOptsImmutable(t *testing.T) {
 	require.True(t, connected, "Reconnect success")
 }
 
+func TestErrorExtendedInfoBasic(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("not a Lua code", []interface{}{})
+	require.NotNilf(t, err, "expected error on invalid Lua code")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "LuajitError",
+		File:  "eval",
+		Line:  uint64(1),
+		Msg:   "eval:1: unexpected symbol near 'not'",
+		Errno: uint64(0),
+		Code:  uint64(32),
+	}
+
+	test_helpers.CheckEqualBoxErrors(t, expected, *ttErr.ExtendedInfo)
+}
+
+func TestErrorExtendedInfoStack(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("error(chained_error)", []interface{}{})
+	require.NotNilf(t, err, "expected error on explicit error raise")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "ClientError",
+		File:  "config.lua",
+		Line:  uint64(214),
+		Msg:   "Timeout exceeded",
+		Errno: uint64(0),
+		Code:  uint64(78),
+		Prev: &BoxError{
+			Type:  "ClientError",
+			File:  "config.lua",
+			Line:  uint64(213),
+			Msg:   "Unknown error",
+			Errno: uint64(0),
+			Code:  uint64(0),
+		},
+	}
+
+	test_helpers.CheckEqualBoxErrors(t, expected, *ttErr.ExtendedInfo)
+}
+
+func TestErrorExtendedInfoFields(t *testing.T) {
+	test_helpers.SkipIfErrorExtendedInfoUnsupported(t)
+
+	conn := test_helpers.ConnectWithValidation(t, server, opts)
+	defer conn.Close()
+
+	_, err := conn.Eval("error(access_denied_error)", []interface{}{})
+	require.NotNilf(t, err, "expected error on forbidden action")
+
+	ttErr, ok := err.(Error)
+	require.Equalf(t, ok, true, "error is built from a Tarantool error")
+
+	expected := BoxError{
+		Type:  "AccessDeniedError",
+		File:  "/__w/sdk/sdk/tarantool-2.10/tarantool/src/box/func.c",
+		Line:  uint64(535),
```

Comment:
The file and the line may change in the future.

Reply:
Yeah, I already have said that `CheckEqualBoxErrors` does not assert exact line and string (https://github.com/tarantool/go-tarantool/pull/229#discussion_r1033435847). But I think its better to have this backtrace + line here as an example of real BoxError.

In `box_error_test.go`, these fields are used on local encode/decode so everything should be alright. In any place where we work with an error from Tarantool instance, we do not assert the file and the line to be exactly equal to ones in example.

Reply:
Oh, sorry, I forgot it. Maybe then do not specify the values in the tests (or add a short note) so that it does not confuse in the future?

Reply:
Added